src/core/execute.c

```diff
@@ -2994,10 +3025,16 @@ static int exec_child(
         }
 
         if (params->cgroup_path) {
-                r = cg_attach_everywhere(params->cgroup_supported, params->cgroup_path, 0, NULL, NULL);
+                _cleanup_free_ char *p = NULL;
+
+                r = exec_parameters_get_cgroup_path(params, &p);
+                if (r < 0)
+                        return log_unit_error_errno(unit, r, "Failed to acquire cgroup path: %m");
```

Comment:
do you also want to set the exit status to `EXIT_CGROUP` here?

Reply:
You are absolutely right, lemme fix that!